pkg/app/ops/insightcollector/application_collector.go

```diff
@@ -59,6 +59,15 @@ func (c *applicationDataCollector) Execute(ctx context.Context) {
 			project = a.ProjectId
 			app     = insight.BuildApplicationData(a)
 		)
+
+		if _, ok := appsByProject[project]; !ok {
+			appsByProject[project] = make([]*insight.ApplicationData, 0)
+		}
+
+		if app.Status == model.ApplicationActiveStatus_DELETED.String() {
+			continue
+		}
```

Comment:
```suggestion
		// Do not count the deleted application.
		if app.Status == model.ApplicationActiveStatus_DELETED.String() {
			continue
		}
```
nits

Reply:
Thanks, added it! [c657a23](https://github.com/pipe-cd/pipecd/pull/4953/commits/c657a2370163be6cfc24b2448b1069ec4ba67d51)